sky/packages/sky/lib/src/rendering/proxy_box.dart

```diff
@@ -126,6 +142,14 @@ class RenderConstrainedBox extends RenderProxyBox {
   String debugDescribeSettings(String prefix) => '${super.debugDescribeSettings(prefix)}${prefix}additionalConstraints: ${additionalConstraints}\n';
 }
 
+/// Forces child to layout at a specific aspect ratio
+///
+/// The width of this render object is the largest width permited by the layout
+/// constraints. The height of the render object is determined by applying the
+/// given aspect ratio to the width.
+///
+/// The aspect ratio is expressed as a ratio of width to height. For example,
+/// a 16:9 width:height aspect ratio would have a value of 16.0/9.0.
```

Comment:
Explain how the layout of this class handles the case where the ratio cannot be honoured when taking the given max width. (For example, if the height and width are both 0..100, but the ratio is 1:2.) Does the width win, or does the ratio win?